modules/xfem/src/userobjects/NodeValueAtXFEMInterface.C

```diff
@@ -7,34 +7,31 @@
 //* Licensed under LGPL 2.1, please see LICENSE for details
```

Comment:
I see what this class is doing after reading your implementation. I think this is fine, but it is better to write a base class called `PointValuesAtXFEMInterface` which only takes a bunch of points, and returns values (and gradients) of a variable at those points -- that could be a very general class.

Then this new class is essentially `CutterMeshNodeValuesAtXFEMInterface` which inherits from `PointValuesAtXFEMInterface` and overrides the way you provide the points. In essence, instead of providing a set of arbitrary points, now you are providing all node coordinates in the cutter mesh.

I think having that base class could be useful in general.